### PR TITLE
Main wrapping

### PR DESCRIPTION
**Pull request by DavidPoliakoff**
This one is blocked by (it uses) #57.

We now handle main wrapping cleanly and transparently. The implementation:

"If I see a main wrap action, (once only) create a thin wrapper around main which merely returns the value of its wrapped function. Redirect libc_start_main to a version which calls that thin wrapper rather than main."

Everything else naturally falls out of that (I think?). If somebody wraps main, they'll get added to the "chain of mains" (dibs on the band name).

I'm open to other implementations, but that one minimally impacts our code. My only concern is if somebody wants to use Gotcha to wrap __libc_start_main, but I'll ask you (@mplegendre) before going too far down that path.

### CODE REVIEW

File: src/translations.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * This file contains utilities for cases where users say something (wrap main) 
+ * that doesn't work, but can be translated to something that does work
+ */
+#ifndef GOTCHA_SRC_TRANSLATIONS_H
+#define GOTCHA_SRC_TRANSLATIONS_H
+#include <gotcha/gotcha.h>
+/** "int main" wrapping handling */
+typedef int (*libc_start_main_t) (int (*)(int, char**, char**), int, char**, void (*)(), void (*)(), void (*)(), void*);
+typedef int (*main_t)            (int argc, char** argv, char** envp);
+static int main_wrapped;
```
Comment (mplegendre) [inline src/translations.h]: Don't define variables in headers.  Declare them with extern, and define them in a .c.

File: test/dlopen/test_dlopen.c [hunk 2]
```diff
@@ -41,9 +41,11 @@ int correct_return_five()
    return 5;
 }
 
-static int (*buggy_return_four)(void);
-static int (*buggy_return_five)(void);
+//static int (*buggy_return_four)(void);
+//static int (*buggy_return_five)(void);
```
Comment (mplegendre) [inline test/dlopen/test_dlopen.c]: Kill comment lines

**Comment by mplegendre** [comment]
@DavidPoliakoff - The variable and table declarations in translations.h still need to be moved to .c files.  See the comments in my previous review.

**Comment by DavidPoliakoff** [comment]
Whoops, worked off the diff from the other PR, will fix that really quick

**Pull request merged**